tests/components/smhi/test_weather.py

```diff
@@ -305,3 +318,35 @@ def get_condition(index: int) -> str:
     assert get_condition(23) == "snowy-rainy"
     # 24. Heavy sleet
     assert get_condition(24) == "snowy-rainy"
+
+
+async def test_custom_speed_unit(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, api_response: str
+) -> None:
+    """Test Wind Gust speed with custom unit."""
+    uri = APIURL_TEMPLATE.format(TEST_CONFIG["longitude"], TEST_CONFIG["latitude"])
+    aioclient_mock.get(uri, text=api_response)
+
+    entry = MockConfigEntry(domain="smhi", data=TEST_CONFIG)
+    entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(ENTITY_ID)
+
+    assert state
+    assert state.name == "test"
+    assert state.attributes[ATTR_SMHI_WIND_GUST_SPEED] == 4.7
+
+    entity_reg = er.async_get(hass)
+    entity_reg.async_update_entity_options(
+        state.entity_id,
+        WEATHER_DOMAIN,
+        {ATTR_WEATHER_WIND_SPEED_UNIT: SPEED_KILOMETERS_PER_HOUR},
+    )
+
+    await asyncio.sleep(2)
```

Comment:
Why do we sleep?

Reply:
The test completes before attributes has refreshed so I made it easy and just put a sleep in.

Reply:
Isnt' it enough to block till done?

Reply:
```suggestion
    await hass.async_block_till_done()
```

Reply:
yes it is...